Add "Show Address" link in State panel

Abyss777 opened the pull request:
If address is empty it is replaced with link to retry geocoding.

![image](https://user-images.githubusercontent.com/5688080/32709391-e3ead1b6-c851-11e7-95cb-2384c2d431d0.png)


Some comments in code

Abyss777 commented on .eslintrc.json:
```diff
@@ -16,7 +16,7 @@
         "max-len": ["error", 180],
         "max-statements": ["error", 35],
-        "max-params": ["error", 5],
+        "max-params": ["error", 7],
```
We need seven parameter in `onCellClick` handler

Abyss777 commented on web/app/view/State.js:
```diff
@@ -63,9 +63,15 @@ Ext.define('Traccar.view.State', {
                 if (record.get('attribute') === 'alarm') {
                     metaData.tdCls = 'view-color-red';
+                } else if (record.get('name') === Strings.positionAddress && !value) {
+                    return '<a href="#">' + Strings.stateShowAddress + '</a>';
```
It is really hard to add `onclick` handler here. (or my knowledge is not enough)
But found on sencha forums tip to handle cell click instead of `a` element.

Abyss777 commented on web/app/view/StateController.js:
```diff
@@ -184,5 +184,30 @@ Ext.define('Traccar.view.StateController', {
+    onCellClick: function (grid, td, cellIndex, record, tr, rowIndex, e) {
+        var positionId = this.position.getId();
+        if (record.get('name') === Strings.positionAddress && !record.get('value') &&
+                e.target.tagName.toLowerCase() === 'a') {
```
We need to determine if user clicked on link (not other place of cell) to leave behavior intuitive.

tananaev replied:
Can we assign some class to the element instead?

Abyss777 commented on web/app/view/StateController.js:
```diff
@@ -184,5 +184,30 @@ Ext.define('Traccar.view.StateController', {
+                    longitude: this.position.get('longitude')
+                },
+                success: function (response) {
+                    if (record && this.position && positionId === this.position.getId()) {
```
This is protection in case request completed after user clicked other position or deselected position.

Abyss777 commented:
I've replaced the commit because I found better solution with help of global events.
Also added the same functionality to route reports

![image](https://user-images.githubusercontent.com/5688080/32714415-9976cee6-c86f-11e7-81bf-cc7e3d5ad3f7.png)

tananaev commented on web/app/view/State.js:
```diff
@@ -63,6 +63,9 @@ Ext.define('Traccar.view.State', {
                     metaData.tdCls = 'view-color-red';
+                } else if (record.get('name') === Strings.positionAddress && !value) {
+                    return '<a href="#" onclick="Ext.fireEvent(\'stategeocode\')" >' +
+                    Strings.stateShowAddress + '</a>';
```
There is an indentation problem here.

tananaev commented on web/app/view/ReportController.js:
```diff
@@ -455,7 +481,14 @@ Ext.define('Traccar.view.ReportController', {
+        renderer: function (value, metaData, record) {
+            if (!value) {
+                return '<a href="#" onclick="Ext.fireEvent(\'routegeocode\', ' +
+                record.getId() + ')" >' +
```
Here as well.

tananaev commented on web/l10n/en.json:
```diff
@@ -241,6 +241,7 @@
     "stateTitle": "State",
     "stateName": "Attribute",
     "stateValue": "Value",
+    "stateShowAddress": "Show Address",
```
I suggest to rename it to `shared*` because it's used in reports as well.

tananaev commented on web/app/view/ReportController.js:
```diff
@@ -423,6 +426,29 @@ Ext.define('Traccar.view.ReportController', {
+                success: function (response) {
+                    position.set('address', response.responseText);
+                    position.commit();
+                    this.fireEvent('selectReport', position);
```
Why do we need to fire this event?

Abyss777 replied:
Clicking on "Show Address" causing selecting the row that causing selecting position and show it in State panel, when comes response address is replaced in Report panel, but not in State.
Looks not very nice. Just updating State panel here.

tananaev replied:
I see. OK.

tananaev commented:
Merged, thanks.

Abyss777 commented:
Sorry for indentation. Pity `eslint` not supports such indentations.
